nemo/core/neural_types/neural_type.py

```diff
@@ -221,6 +221,7 @@ def __init__(self, producer, producer_args, output_port_name, ntype=None):
         self._producer_args = producer_args
         self._output_port_name = output_port_name
         self._name = output_port_name
+        self._output_port_name = output_port_name
```

Comment:
hey, so it seems that those two properties store exactly the same information...

I guess you want to leave self._name (and name property) to keep things _compatible_..?

Reply:
No they are not the same. Do the docstrings of the properties help you differentiate between their uses?
```
    @property
    def name(self):
        """
        Returns:
            A NmTensor's name. By default, it is equal to self.output_port_name, but can be renamed using self.rename.
            Used in self.__str__() to obtain a string representation of a nmtensor.
        """
        return self._name
```
```
    @property
    def output_port_name(self):
        """
        Returns:
            The name of the output port it is associated with. This is used in DAG creation.
        """
        return self._output_port_name
```

If they do not, please let me know because I should make it more clear.

Reply:
Ok, I see. My point was that in init():
```
        self._name = output_port_name
        self._output_port_name = output_port_name
```
i.e. they have the same value set.

But now I've read the description, seems that the `.rename` option allows them to be "decoupled"...